File: 55. Jump Game.java
class Solution {
    public boolean canJump(int[] nums) {
        int n = nums.length;
        int i = 0;
        int maxDistant = nums[0];

        for(; i < n && i <= maxDistant; ++i){
            if(maxDistant < nums[i] + i) maxDistant = nums[i] + i;
        }

        return n == i;
    }
}

File: 55. Jump Game.py
class Solution(object):
    def canJump(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        maxDistant = nums[0]
        i = 0

        while(i < len(nums) and i <= maxDistant):
            if(maxDistant < i + nums[i]):
                maxDistant = i + nums[i]
            i += 1

        return i == len(nums)